app/main.py

```diff
@@ -1,52 +1,36 @@
-import json
-import xml.etree.ElementTree as ET
-
-
 class Book:
-    def __init__(self, title: str, content: str):
+    def __init__(self, title: str, content: str) -> None:
         self.title = title
         self.content = content
 
-    def display(self, display_type: str) -> None:
-        if display_type == "console":
-            print(self.content)
-        elif display_type == "reverse":
-            print(self.content[::-1])
-        else:
-            raise ValueError(f"Unknown display type: {display_type}")
-
-    def print_book(self, print_type: str) -> None:
-        if print_type == "console":
-            print(f"Printing the book: {self.title}...")
-            print(self.content)
-        elif print_type == "reverse":
-            print(f"Printing the book in reverse: {self.title}...")
-            print(self.content[::-1])
-        else:
-            raise ValueError(f"Unknown print type: {print_type}")
-
-    def serialize(self, serialize_type: str) -> str:
-        if serialize_type == "json":
-            return json.dumps({"title": self.title, "content": self.content})
-        elif serialize_type == "xml":
-            root = ET.Element("book")
-            title = ET.SubElement(root, "title")
-            title.text = self.title
-            content = ET.SubElement(root, "content")
-            content.text = self.content
-            return ET.tostring(root, encoding="unicode")
-        else:
-            raise ValueError(f"Unknown serialize type: {serialize_type}")
-
 
 def main(book: Book, commands: list[tuple[str, str]]) -> None | str:
+    from app.serializers import JSONBookSerializer, XMLBookSerializer
+    from app.displayers import ConsoleDisplayer, ReverseDisplayer
+    from app.printers import ConsolePrinter, ReversePrinter
+
     for cmd, method_type in commands:
         if cmd == "display":
-            book.display(method_type)
+            if method_type == "console":
+                displayer = ConsoleDisplayer()
+                return displayer.display(book)
+            elif method_type == "reverse":
+                displayer = ReverseDisplayer()
+                return displayer.display(book)
         elif cmd == "print":
-            book.print_book(method_type)
+            if method_type == "console":
+                printer = ConsolePrinter()
+                return printer.print(book)
+            elif method_type == "reverse":
+                printer = ReversePrinter()
+                return printer.print(book)
         elif cmd == "serialize":
-            return book.serialize(method_type)
+            if method_type == "json":
+                serializer = JSONBookSerializer()
+                return serializer.serialize(book)
+            elif method_type == "xml":
+                serializer = XMLBookSerializer()
+                return serializer.serialize(book)
```

Comment:
Too many ifs in one place. I'd recommend to create some functions like display_factor, print_factory, etc that you'll always update when a new Printer or Displayer is created. It'll be easier to read the idea of main function.